server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -624,6 +624,12 @@ public static IndexMergePolicy fromString(String text) {
         Property.IndexScope
     );
 
+    public static final Setting<Boolean> INDEX_CHECK_PENDING_FLUSH_ENABLED = Setting.boolSetting(
+        "index.check_pending_flush.enabled",
+        true,
+        Property.IndexScope
```

Comment:
Please add
```
        Property.Dynamic,
```

since you allow changing it dynamically

Reply:
@reta this is a static setting only which requires closing of index first. There's a bug with few other settings like `INDEX_MERGE_ON_FLUSH_ENABLED` which are [static settings](https://opensearch.org/docs/latest/install-and-configure/configuring-opensearch/index-settings/#static-index-level-index-settings) but are marked Dynamic in code

Reply:
+1 
I had created an issue to discuss this: https://github.com/opensearch-project/OpenSearch/issues/10048